src/nats.h

```diff
@@ -186,6 +186,28 @@ typedef struct __natsOptions        natsOptions;
  */
 typedef char                        natsInbox;
 
+
+/** \brief An initial configuration for NATS client. Provides control over the
+ * threading model, and sets many default option values.
+ *
+ * @see natsOptions_Create
+ */
+typedef struct __natsClientConfig
+{
+        int64_t DefaultWriteDeadline;
```

Comment:
This would be the "role model" for the future:
A `ClientConfig` that we can accept as JSON, used as the default, overridable with env. variables. In this PR I just wanted finer control over the thread pool than the prior ENV variable.